### PR TITLE
Fix InputLanguage null refs and add tests

### PR DESCRIPTION
**Pull request by hughbe**
Fixes https://github.com/dotnet/winforms/issues/379
Fixes https://github.com/dotnet/winforms/issues/378

Also some code cleanup

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs [hunk 1]
```diff
@@ -244,135 +225,131 @@ Language ID
...
+        /// Specifies whether two input languages are equal.
         /// </devdoc>
-        public override bool Equals(object value) {
-            if (value is InputLanguage) {
-                return(this.handle == ((InputLanguage)value).handle);
-            }
-            return false;
+        public override bool Equals(object value)
+        {
+            return value is InputLanguage other && this.handle == other.handle;
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs]: nit: can you parenthesize both of these checks for clarity

```suggestion
            return (value is InputLanguage other) && (this.handle == other.handle);
```

File: src/System.Windows.Forms/src/System/Windows/Forms/InputLanguageChangedEventArgs.cs [hunk 2]
```diff
@@ -28,6 +28,11 @@ public InputLanguageChangedEventArgs(CultureInfo culture, byte charSet)
         /// </devdoc>
         public InputLanguageChangedEventArgs(InputLanguage inputLanguage, byte charSet)
         {
+            if (inputLanguage == null)
+            {
+                throw new ArgumentNullException(nameof(inputLanguage));
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/InputLanguageChangedEventArgs.cs]: What would be the issue here if this were null? Should this be thrown in the constructor or from the funciton seeking to use InputLanguage which is null?

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/InputLanguageChangedEventArgs.cs]: The issue below is that we have `Culture = inputLanguage.Culture;` so this throws NRE

File: src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs [hunk 3]
```diff
@@ -244,135 +225,131 @@ Language ID
...
-        ///    <para>Returns the input language associated with the specified 
-        ///       culture.</para>
+        /// Returns the input language associated with the specified culture.
         /// </devdoc>
-        public static InputLanguage FromCulture(CultureInfo culture) {
+        public static InputLanguage FromCulture(CultureInfo culture)
+        {
+            if (culture == null)
+            {
+                throw new ArgumentNullException(nameof(culture));
```
Comment (KlausLoeffelmann) [inline src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs]: What happens to big monolithic software which gets migrated to core, which somewhere in its 1MLOC+ catches NullReferenceExceptions which ArgumentNullException is not derived from?

File: src/System.Windows.Forms/src/System/Windows/Forms/InputLanguageChangedEventArgs.cs (hunk 2 above)
Comment (KlausLoeffelmann) [inline src/System.Windows.Forms/src/System/Windows/Forms/InputLanguageChangedEventArgs.cs]: Same here.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageChangedEventArgsTests.cs [hunk 4]
```diff
@@ -51,15 +51,15 @@ public void Ctor_InputLanguage_Byte(InputLanguage inputLanguage, byte charSet)
 #endif
 
         [Fact]
-        public void Ctor_NullCultureInfo_ThrowsNullReferenceException()
+        public void Ctor_NullCultureInfo_ThrowsArgumentNullException()
         {
-            Assert.Throws<NullReferenceException>(() => new InputLanguageChangedEventArgs((CultureInfo)null, 0));
```
Comment (KlausLoeffelmann) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageChangedEventArgsTests.cs]: Exactly my point.

File: src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs (hunk 3 above)
Comment (danmosemsft) [reply src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs]: FWIW, in dotnet/corefx we have taken quite a few of these changes. It is borderline but we generally consider them acceptably low breaking risk. I take your point that there may be cases where a misbehaving library may pass null and a client is working around it by catching NRE but in practice, so far we have not gotten negative feedback. This is the only case I can think of where we have been OK with changing exception type.
That's in corefx though, winforms can certainly have its own policy, has different clients, etc.

Comment (danmosemsft) [reply src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs]: And indeed @hughbe has been the one who made many of these changes 😼

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs]: Waiting on https://github.com/dotnet/winforms/issues/392

**Comment by hughbe** [comment]
@zsd4yr now we’ve reached new consensus to take these fixes, this is ready to be reviewed

**Comment by zsd4yr** [comment]
LGTM

**Pull request merged**